python/jupyterlab-chat/jupyterlab_chat/models.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+from dataclasses import dataclass
+from typing import Literal, Optional
+from jupyter_server.auth import User as JupyterUser
+
+
+@dataclass(kw_only=True)
+class Message:
+    """ Object representing a message """
+
+    type: Literal["msg"] = "msg"
+    body: str
+    """ The content of the message """
+
+    id: str
+    """ Unique ID """
+
+    time: float
+    """ Timestamp in second since epoch """
+
+    sender: str
+    """ The message sender unique id """
+
+    raw_time: Optional[bool] = None
+    """ Whether the timestamp is raw (from client) or not (from server, unified) """
+
+    deleted: Optional[bool] = None
+    """ Whether the message has been deleted or not (body should be empty if True) """
+
+    edited: Optional[bool] = None
+    """ Whether the message has been edited or not """
+
+
+@dataclass(kw_only=True)
+class User(JupyterUser):
+    """ Object representing a user (same as Jupyter User ) """
```

Comment:
Subclassing this allows for attributes to be added in the future, extending the `User` model provided by Jupyter Server. Good thinking!